openeo/extra/job_management.py

```diff
@@ -202,6 +209,39 @@ def _normalize_df(self, df: pd.DataFrame) -> pd.DataFrame:
         df = df.assign(**new_columns)
 
         return df
+    
+    def _check_and_stop_long_running_jobs(self, df: pd.DataFrame):
```

Comment:
I'm wondering if it is necessary to define a separate method for this actually.

I think you can  do this cancelling in the existing for-loop of `_update_statuses`, instead of duplicating a bunch of stuff

Reply:
I would do the canceling from within `_update_statuses`.

Note that `_update_statuses` is about more than updating statuses anyway: it triggers `on_job_done` and `on_job_error`, which might be non-trivial things. I think `_update_statuses` is a bit of misnomer at the moment

Reply:
Sounds good, should I rename it to _update_jobs within this PR?

Reply:
`_tracks_statuses`  makes more sense at the moment I guess